path_collector.go

```diff
@@ -115,7 +116,13 @@ func (c *DoublestarCollector) CollectPaths() ([]string, error) {
 		}
 		excluded := false
 		for _, pattern := range c.Exclude {
-			match, err := doublestar.Match(filepath.Clean(pattern), path)
+			absPath, err := filepath.Abs(path)
+			if err != nil {
+				// I wonder how this could ever happen...
+				log.Printf("could not create absolute path for %s: %v", path, err)
```

Comment:
This literally never happens on a day-to-day basis, but an error can trigger under some improbable circumstances
Internally `filepath.Abs` calls the `getwd` syscall, which has quite a few possible errors : https://linux.die.net/man/3/getwd

Here is a small example triggering an error on purpose : https://go.dev/play/p/JoPt8x9bQJl

Reply:
Ah good to know, thank you for the information!

I think I am okay with how I'm handling this for now then; if it was able to collect that path in the first place up until this point, then failing `getwd` doesn't necessarily mean the program can't eventually read that file from the relative path it currently has. So logging the error and moving on will at least notify the user that if they expect that path to be excluded, something unexpected went wrong.

(I probably should have thrown a newline on that log though...)